api/v1alpha3/openstackcluster_types.go

```diff
@@ -115,6 +115,9 @@ type OpenStackClusterSpec struct {
 	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
 	// +optional
 	ControlPlaneEndpoint clusterv1.APIEndpoint `json:"controlPlaneEndpoint"`
+
+	// AvailabilityZone is the az to deploy controller to
+	AvailabilityZone string `json:"availabilityZone,omitempty"`
```

Comment:
Why do we need an additional property? Why not just set the az on the OpenStackMachineTemplate of KubeadmControlPlane like for MachineDeployments? I think that would be much more consistent. The property is also already there you just have to use it

Reply:
Maybe a compromise would be to make it an AZ array. So this means if you have e.g. 5 AZs and you only want to distribute the control plane to 3 of them you can choose which (of course in your case it would be just one)

Reply:
I had some analysis at #525
I think the az are returned by openstack side and cluster-api side seems don't have spec for this
maybe you can point me where should I add? The machine is created by cluster-api side automatically with the param found and I don't know it will honor the OpenStackMachineTemplate settings we give...

I think the az will be a good suggestion, thanks

Reply:
Oh sorry of course you're right. I removed the az property in favor of the failureDomain property in the CAPI Machine resource :) Ups

So yes you're solution is the only way to restrict the AZs. Hm maybe we should open an issue in the main ClusterAPI repo to ask if they want to provide a way to restrict the AZs via a property in the KubeAdmControplane, might be an oversight that this cannot be specified there.

Reply:
let's ask there (I will do it soon) before we do this and see whether they have better suggestions :)

Reply:
Okay perfect :). If they don't want to do anything there, I'm fine with solving it in CAPO.